src/engine/channels/enginedeck.cpp

```diff
@@ -39,39 +46,110 @@ EngineDeck::EngineDeck(
             pConfig,
             this,
             pMixingEngine,
+#ifdef __STEM__
             primaryDeck ? mixxx::audio::ChannelCount::stem()
-                        : mixxx::audio::ChannelCount::stereo());
+                        : mixxx::audio::ChannelCount::stereo()
+
+#else
+            mixxx::audio::ChannelCount::stereo()
+#endif
+    );
+
+#ifdef __STEM__
+    if (!primaryDeck) {
+        return;
+    }
```

Comment:
No, this is an early return as we don't want to add stem controls on non-primary deck (sampler and preview), since they are not stem-capable due to performance reason. As part of #13268, non-primary deck will be able to load individual channels.